### DISCUSSION TITLE
File level diagnostic config only applies to current cell for jupyter notebook in vscode

### DISCUSSION THREAD

**Discussion by colinfang** [microsoft/pylance-release#4499] [Enhancement]
## Environment data

-   Language Server version: v2023.6.10 
-   OS and version: WSL2
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11

Here I put `# pyright: reportUnusedImport=none` on the first cell yet the lint still warns about the same error.

It seems the file level config only applies to the current cell, rather than all cells below.

![image](https://github.com/microsoft/pylance-release/assets/1499555/3652bf8b-8acd-442f-bb42-6184a7347335)

**Comment by debonte** [comment]
That makes sense given the implementation -- each notebook cell is a separate document from Pylance's perspective.

We could probably make this work. Might want to wait for upvotes before investing time in it though.
